.github/workflows/compilation.yml

```diff
@@ -8,53 +8,72 @@ on:
 
 jobs:
   build:
-    runs-on: ${{ matrix.os }}
+    runs-on: ${{ matrix.os[0] }}
     strategy:
       matrix:
-        os: [macos-latest, ubuntu-latest]
+        os: [[macos-latest, bash], [ubuntu-latest, bash], [windows-latest, msys2]]
+      fail-fast: false
+    defaults:
+     run:
+      shell: ${{ matrix.os[1] }} {0}
 
     steps:
     - uses: actions/checkout@v2
       with:
         fetch-depth: 0
     
     - name: Install Ubuntu texinfo bison flex libucl-dev
-      if: matrix.os == 'ubuntu-latest'
+      if: matrix.os[0] == 'ubuntu-latest'
       run: |
         sudo apt-get update
         sudo apt-get -y install texinfo bison flex gettext libgmp3-dev libmpfr-dev libmpc-dev
+        echo "MSYSTEM=x64" >> $GITHUB_ENV
     
     - name: Install Mac texinfo bison flex ucl
-      if: matrix.os == 'macOS-latest'
+      if: matrix.os[0] == 'macOS-latest'
       run: |
         brew update
         brew install texinfo bison flex gnu-sed gsl gmp mpfr
+        echo "MSYSTEM=x64" >> $GITHUB_ENV
+
+    - name: Install MSYS2 texinfo bison flex
+      if: matrix.os[0] == 'windows-latest'
+      uses: msys2/setup-msys2@v2
+      with:
+        msystem: MINGW32
+        install: |
+          base-devel git make texinfo flex bison patch binutils mingw-w64-i686-gcc mpc-devel tar
+          mingw-w64-i686-cmake mingw-w64-i686-extra-cmake-modules mingw-w64-i686-make
+        update: true
+        shell: msys2 {0}
 
     - name: Runs all the stages in the shell
+      continue-on-error: true
       run: |
         export PS2DEV=$PWD/ps2dev
         export PS2SDK=$PS2DEV/ps2sdk
         export GSKIT=$PS2DEV/gsKit
         export PATH="/usr/local/opt/gnu-sed/libexec/gnubin:$PATH"
         export PATH=$PATH:$PS2DEV/bin:$PS2DEV/ee/bin:$PS2DEV/iop/bin:$PS2DEV/dvp/bin:$PS2SDK/bin
-        ./build-all.sh
+        # ./build-all.sh
```

Comment:
This can be changed when ps2client will be fixed

Reply:
I think that is cleaner to modify `006-ps2client-extra.sh` and put there the condition `if MINGW .... exit`

Reply:
Its possible, but is it so crucial to have ps2client inside the toolchain?

Reply:
I managed to fix ps2client